packages/ingest-mongodb-public/src/sources/mongodb-university/MongoDbUniversityDataApiClient.ts

```diff
@@ -157,12 +175,35 @@ export function makeMongoDbUniversityDataApiClient({
   };
 
   return {
-    async getAllCatalogItems() {
-      const response = await fetch(`${baseUrl}/ti`, {
+    async getCatalogItems({
+      publicOnly,
+      learningFormats,
+      nestAssociatedContent,
+    }: {
+      publicOnly?: boolean;
+      learningFormats?: string[];
+      nestAssociatedContent?: boolean;
+    } = {}) {
+      const search_params = new URLSearchParams();
+      if (publicOnly !== undefined) {
```

Comment:
I wrote it that way to give us the option to explicitly set publicOnly to false. For example, if we call getCatalogItems like this
```
 await mongodbUniversityDataApiClient.getCatalogItems({
          publicOnly: false,
          nestAssociatedContent: false,
});
```
the url will look like `catalog/ti?public_only=false&nest_associated_content=false`, explicitly setting those fields to false, rather than relying on whatever defaults the catalog/ti endpoint has. Currently the default is false for both of theses fields so maybe this is unnecessarily complicated and confusing.

Reply:
ahh ok, i should have payed closer attention to the code inside the if statement. lgtm

Reply:
I found a way to do this with defaults where we can still explicitly set false and went with that